### PR TITLE
NEW batch deleting for products and services in proposals, billings and client's orders

### PR DESCRIPTION
**Pull request by air-dex**
Batch deleting for products and services in proposals, billings and client's orders with checkboxes.

It is a feature that we at Code Couleurs have developed for our own Dolibarr install and that we want to share with Dolibarr's upstream.

### CODE REVIEW

**Comment by rdoursenaud** [comment]
You are a bit late to the party since we just announced feature freeze a few days ago.

Hope this can be merged in the next round.

File: htdocs/comm/propal.php [hunk 1]
```diff
@@ -182,6 +182,49 @@
 	}
 }
 
+// Suppression de plusieurs services
```
Comment (rdoursenaud) [inline htdocs/comm/propal.php]: Comments in english please.

**Comment by rdoursenaud** [comment]
You also have a few cosmetic mistakes regarding Dolibarr coding style. See: https://travis-ci.org/Dolibarr/dolibarr/jobs/25232416#L2401

**Comment by BebZ** [comment]
Wow ! THIS is a big piece of (art)work ;)
I hope you have time and a strong lobby to make it voted in the "Brussels of Dolibarr" ;)
I am your €U deputy to make it adopted in Dolibarr's constitution :)

**Comment by BebZ** [comment]
No more news on this pull request ? It would be really handy to have this kind of implementation...
I'm spending hours deleting lines sometimes...
Wherever there is a "product list" would it be possible to make like "mass actions" (exemple : out of sale, out of purchasing, define same custom field for selected refs, add selected refs to PROVn°, etc...) ? Yes, no, one day ?

**Comment by BebZ** [comment]
Hello, here is a good piece of work, something useful for Dolibarr, making it twice more user-friendly !
Pleaaaase can you add it on the roadmap for 3.8 ?

**Comment by BebZ** [comment]
Please make a pull request on develop maybe some day someone will notice it???

**Comment by BebZ** [comment]
Air-dex are you still here or disgusted ?

**Comment by altairis-tof** [comment]
hello, i did not look at the code, but i don't understand why there are no answers to these interrogations... many users are awaiting for this functionality

**Comment by air-dex** [comment]
@BebZ Nope. I am still here. I follow the conflicts through mail notifications about it.

I thought that you do not need me about this pull request so that's why I did not take part in this conversation. So what I need to do about it?

**Comment by BebZ** [comment]
Thanks for your answer ! 
He was asking you to Pull Request it on the Develop branch of Dolibarr, and to see the cosmetics mistakes of Travis (i don't understand it as 'im not a dev...)
https://travis-ci.org/Dolibarr/dolibarr/jobs/25232416#L2401

do you have a working demo version of this multiple selection I can see ?

**Comment by BebZ** [comment]
@air-dex have you seen my answer ?
Can you make a pull request on the develop branch of dolibarr before its too late ?

**Comment by air-dex** [comment]
I am on it. I will likely make the new pull request tomorrow (Wednesday) or Thrusday. There are months of new commits on the Dolibarr's upstream since this original pull request and I want to test them before making the new pull request.

**Comment by air-dex** [comment]
@BebZ Done! The pull request was originally made on the develop branch so I did not need to make another one. I merged the lastest Dolibarr's commits (3983 commits since May 2014) in the pull request's branch (code-couleurs/dolibarr:delete_lot) and tested it to ensure that it works.

Special thanks to @emarguin ( https://github.com/emarguin ) to allow me to do this work on the pull request.

**Comment by BebZ** [comment]
@eldy so here it is... maybe now is not late anymore to the party for 3.8 ? @rdoursenaud

**Comment by BebZ** [comment]
smthg wrong with PR ?

**Comment by rdoursenaud** [comment]
@BebZ I think you have to be a little bit more patient. The todo tag certainly means that @eldy needs some time to review the feature and will do it when he's available.

**Comment by BebZ** [comment]
:( and the label feature request, means what ? to be discussed for 3.8, or to be integrated soon or later ?

**Comment by eldy** [comment]
@BebZ the label just mean the PR is a new feature. So i don't loose time next time to know this. Depending on project planning, bugs and features are not processed with same priority.

**Comment by BebZ** [comment]
Was this feature request discussed in the DevCamp 2015 ?
Batch selection of products is something useful isn't it ?
or doest it have too much implications on code and the balance on benefits/effort/risk is not worth it ?.

**Comment by BebZ** [comment]
How can I test it ? I just take the modified files and inject in my local environement ?
On a latest 3.8 or on a 3.7 ?

**Comment by rdoursenaud** [comment]
@BebZ The best way is that you learn how to use git.
This way, you'll be able to clone this branch from the PR submitter's repository and safely test it (provided you also backup your database).

_Resources:_
http://try.github.io
https://training.github.com/kit/downloads/fr/github-git-cheat-sheet.pdf
http://git-scm.org

**Comment by eldy** [comment]
Another tutorial dedicated to dolibarr project is also here:
http://wiki.dolibarr.org/index.php?title=FAQ_Get,update_GIT_project_sources&action=history

Because this feature may break or not be compatible with the mutualized tpl file for managing lines introduced into 3.7, i am waiting more feedback to have report this does not break dolibarr with develop branch before merging it.

**Comment by eldy** [comment]
With version 3.8, i have redesigned the way table are rendered. Now, the status is not the last column, but penultimate. And into last column, we will found the loop to refrech a search into title bar and a checkbox into each lines of tables.
Checking this checkbox, will offer a way to execute a mass action. So i think best think is still to wait and recode this part, it will be easier to have such feature after 3.8

**Comment by BebZ** [comment]
Whoa! i didnt see the checkbox yet; but it's a real opened path to mass actions ! 
Thanks Eldy; and thanks in advance to the guy who will write the functions for mass actions (delete; add to proposal xxx; add to invoice xxx; add to mass stock movemets, etc..)

**Pull request closed without merging**